provision/kubernetes/provisioner.go

```diff
@@ -640,11 +640,19 @@ func setNodeMetadata(node *apiv1.Node, pool, iaasID string, meta map[string]stri
 	}
 	for k, v := range meta {
 		k = tsuruLabelPrefix + strings.TrimPrefix(k, tsuruLabelPrefix)
-		if v == "" {
-			delete(node.Annotations, k)
-		} else {
+		switch k {
+		case tsuruExtraAnnotationsMeta:
+			appendKV(v, ",", "=", node.Annotations)
+		case tsuruExtraLabelsMeta:
+			appendKV(v, ",", "=", node.Labels)
+		default:
+			if v == "" {
```

Comment:
Shouldn't the default section be actually outside the switch statement to ensure the annotation `tsuru.io/extra-labels,annotation` itself is added to the node?